src/hardware/vga_s3.cpp

```diff
@@ -822,4 +829,106 @@ void SVGA_Setup_S3Trio(void)
 
 	const auto num_modes = ModeList_VGA.size();
 	VGA_LogInitialization(description.c_str(), ram_type.c_str(), num_modes);
+
+#if defined(PCI_FUNCTIONALITY_ENABLED)
+	PCI_AddSVGAS3_Device();
+#endif
 }
+
+#if defined(PCI_FUNCTIONALITY_ENABLED)
+
+struct PCI_VGADevice : public PCI_Device {
+	enum { vendor = 0x5333 }; // S3
+	enum { device = 0x8811 }; // trio64
+	//enum { device = 0x8810 }; // trio32
+
+	PCI_VGADevice():PCI_Device(vendor,device) { }
+
+	Bits ParseReadRegister(uint8_t regnum)
+	{
+		return regnum;
+	}
+
+	bool OverrideReadRegister(uint8_t regnum, uint8_t* rval, uint8_t* rval_mask)
+	{
+		return false;
+	}
+
+	Bits ParseWriteRegister(uint8_t regnum, uint8_t value)
```

Comment:
Make all args const where applicable, remove the `Bits` stuff, etc.

Reply:
All Bit* changes so far were automatic find/replace; for Bits I will actually need to think if the replacement is ok ;) Will do.